### PR TITLE
Add management command for repopulating subscriptions

### PR DESCRIPTION
**Pull request by KaitCrawford**
There have been a number of bugs that have prevented registrations from being converted into subscriptions (the most recent being incorrect message set names). These bugs have been fixed and this PR adds a management command to re-run the validation on registrations that don't have subscriptions. This will result in the subscriptions being created.

### CODE REVIEW

**Comment by KaitCrawford** [comment]
@smn  please review

File: registrations/management/commands/repopulate_subscriptions.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                              'already has subscription (identity: %s). '
+                              'Skipping.')
+                             % (reg.pk, reg.mother_id))
+                    continue
+
+            """
+            validate_registration() ensures no invalid registrations get
+            subscriptions and creates the Subscription Request
+            """
+            output = validate_registration(registration_id=str(reg.id))
```
Comment (rudigiesler) [inline registrations/management/commands/repopulate_subscriptions.py]: I think this needs to be indented to be in the for loop.

**Review by rudigiesler — COMMENTED** [review]
Could we also have a unit test for this management command? You can use https://docs.djangoproject.com/en/1.10/ref/django-admin/#django.core.management.call_command

File: registrations/management/commands/repopulate_subscriptions.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    'Please make sure either the STAGE_BASED_MESSAGING_TOKEN '
+                    'environment variable or --sbm-token is set.')
+            client = StageBasedMessagingApiClient(sbm_token, sbm_url)
+
+        registrations = Registration.objects.filter(validated=True)
+
+        for reg in registrations:
+            requests = reg.get_subscription_requests()
+            if requests.count() > 0:
+                continue
```
Comment (smn) [inline registrations/management/commands/repopulate_subscriptions.py:64]: PostgreSQL is _phenomenally_ bad at counting things. It's always better using `.exists()` than `count()` as the existence for at least 1 record is always cheaper than doing a count of exactly how many there are.

File: registrations/management/commands/repopulate_subscriptions.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            for req in requests:
+                if check_subscription and self.count_subscriptions(client,
+                                                                   req):
+                    self.log(('Registration %s without Subscription Requests '
+                              'already has subscription (identity: %s). '
+                              'Skipping.')
+                             % (reg.pk, reg.mother_id))
+                    continue
```
Comment (smn) [inline registrations/management/commands/repopulate_subscriptions.py]: I probably should now this but I'm not sure if this breaks us out of both `for` loops here.

**Review by smn — COMMENTED** [review]
Looks good, just let a few minor comments. It would be good to have a test for this. Shout if you need help getting started on that, Django has a `call_command` method that you can use to test these kinds of things.

**Comment by KaitCrawford** [comment]
@rudigiesler @smn ready for re-review

File: registrations/test_commands.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+
+class ManagementCommandsTests(AuthenticatedAPITestCase):
+    def test_command_requires_sbm_url(self):
+        with self.assertRaises(management.CommandError) as ce:
+            management.call_command("repopulate_subscriptions")
+        self.assertEqual(
+            str(ce.exception), "Please make sure either the "
+            "STAGE_BASED_MESSAGING_URL environment variable or --sbm-url is "
+            "set.")
```
Comment (smn) [inline registrations/test_commands.py:20]: hah, never thought of doing this. Excellent :) :cake:

**Review by smn — APPROVED** [review]

**Pull request merged**